app/questionnaire/path_finder.py

```diff
@@ -18,9 +18,11 @@ def __init__(
         list_store: ListStore,
         progress_store: ProgressStore,
         metadata: Mapping,
+        response_metadata: dict,
```

Comment:
We have some inconsistencies in where we used dict vs mapping, for these I think we should change to `Mapping`. You will to change the typing in value source from `dict` to `Mapping` as well.